habitat-lab/habitat/tasks/rearrange/utils.py

```diff
@@ -431,7 +437,53 @@ def place_robot_at_closest_point(
             f"Object {target_position} is out of bounds but trying to set robot position to {agent_pos}"
         )
 
+    # Get the normal agent position
+    agent_pos = sim.safe_snap_point(target_position)
     desired_angle = get_angle_to_pos(np.array(target_position - agent_pos))
+    # Cache the initial location of the agent
+    cache_pos = sim.articulated_agent.base_pos
+
+    # Make a copy of agent trans
+    trans = mn.Matrix4(sim.articulated_agent.sim_obj.transformation)
+
+    accept_agent_center = []
+
+    if navmesh_offset is not None:
```

Comment:
Put this in a separate method for cleaner code.

Reply:
Thank you, done!